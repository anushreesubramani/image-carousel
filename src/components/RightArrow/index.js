import React from "react";
import arrow from "assets/arrow.svg";
import { RightArrowWrapper } from "./styles";

const RightArrow = props => {
  return (
    <RightArrowWrapper data-testid="rightclick" onClick={props.goToNextSlide}>
      <img src={arrow} alt="rightarrow" />
    </RightArrowWrapper>
  );
};

export default RightArrow;
