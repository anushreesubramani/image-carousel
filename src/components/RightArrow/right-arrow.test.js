import React from "react";
import { render, cleanup, fireEvent } from "react-testing-library";
import RightArrow from "components/RightArrow";

afterEach(cleanup);

describe("testing left arrorw", () => {
  it("rendering right arrow and click", () => {
    const handleClick = jest.fn();
    const { getByAltText, getByTestId, container } = render(
      <RightArrow goToNextSlide={handleClick} />
    );
    const arrowElement = getByTestId("rightclick");
    fireEvent.click(arrowElement);
    fireEvent.click(arrowElement);
    expect(handleClick).toHaveBeenCalledTimes(2);
    expect(getByAltText("rightarrow").alt).toBe("rightarrow");
    expect(container).toMatchSnapshot();
  });
});
