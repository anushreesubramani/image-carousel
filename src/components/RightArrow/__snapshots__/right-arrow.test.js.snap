// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing left arrorw rendering right arrow and click 1`] = `
<div>
  <div
    class="sc-bdVaJa ePgJGs"
    data-testid="rightclick"
  >
    <img
      src="arrow.svg"
      alt="rightarrow"
    />
  </div>
</div>
`;
